src/main/java/com/microsoft/aad/adal4j/AuthenticationContext.java

```diff
@@ -282,6 +283,93 @@ public AuthenticationResult call() throws Exception {
+
+                    // Make the OAuth2 call to get the access Token.
+                    result = acquireTokenCommon(authGrant, this.clientAuth, this.headers);
+                    logResult(result, headers);
+                    if (callback != null) {
+                        callback.onSuccess(result);
+                    }
+                }
+                catch (final Exception ex) {
+                    log.error(LogHelper.createMessage("Request to acquire token failed.", this.headers.getHeaderCorrelationIdValue()), ex);
+                    if (callback != null) {
+                        callback.onFailure(ex);
+                    }
+                    else {
+                        throw ex;
+                    }
+                }
+                return result;
+            }
+
+            private Callable<AuthenticationResult> init(String userName,
+                    final String resource,
+                    final ClientAuthentication clientAuth,
+                    final ClientDataHttpHeaders headers) {
+                this.userName = userName;
+                this.resource = resource;
+                this.clientAuth = clientAuth;
+                this.headers = headers;
+                return this;
+            }
+
+        }.init(userName, resource, clientAuth, new ClientDataHttpHeaders(this.getCorrelationId())));
+    }
+
+    AuthorizationGrant getAuthorizationGrantIntegrated(String userName) throws Exception {
+        AuthorizationGrant updatedGrant = null;
+
+        String userRealmEndpoint = authenticationAuthority.getUserRealmEndpoint(URLEncoder.encode(userName, "UTF-8"));
+
+        // Get the realm information
+        UserDiscoveryResponse userRealmResponse = UserDiscoveryRequest.execute(userRealmEndpoint, proxy, sslSocketFactory);
+
+        if (userRealmResponse.isAccountFederated() && "WSTrust".equalsIgnoreCase(userRealmResponse.getFederationProtocol())) {
+            String mexURL = userRealmResponse.getFederationMetadataUrl();
+            String cloudAudienceUrn = userRealmResponse.getCloudAudienceUrn();
+
+            // Discover the policy for authentication using the Metadata Exchange Url.
+            // Get the WSTrust Token (Web Service Trust Token)
+            WSTrustResponse wsTrustResponse = WSTrustRequest.execute(mexURL, cloudAudienceUrn, proxy, sslSocketFactory);
+
+            if (wsTrustResponse.isTokenSaml2()) {
+                updatedGrant = new SAML2BearerGrant(new Base64URL(Base64.encodeBase64String(wsTrustResponse.getToken().getBytes("UTF-8"))));
+            }
+            else {
+                updatedGrant = new SAML11BearerGrant(new Base64URL(Base64.encodeBase64String(wsTrustResponse.getToken().getBytes())));
+            }
+        }
+
+        return updatedGrant;
+    }
+
+    public Future<AuthenticationResult> acquireTokenIntegrated(String userName,

```

Comment:
in order to be consistent with other Adal libraries we should not add new Api for integ Auth, instead existing one ( acquireToken(final String resource, final String clientId, final String username, final String password, final AuthenticationCallback callback) ) should be used, so if it is called with password = null It should be treated as Integrated auth case

Reply:
sure. Do you want me to use (password == null) or (StringHelper.isBlank(password))? Please let me know. Otherwise, I will continue with (password == null) and skip the case of "". Thank you.